ext/webgpu/02_surface.js

```diff
@@ -157,8 +158,28 @@ function createCanvasContext(options) {
   return canvasContext;
 }
 
-function presentGPUCanvasContext(ctx) {
-  ctx[_present]();
+// External webgpu surfaces
+
+// TODO(@littledivy): This will extend `OffscreenCanvas` when we add it.
+class UnsafeWindowSurface {
+  #ctx;
+  #surfaceRid;
+
+  constructor(system, win, display) {
+    this.#surfaceRid = ops.op_webgpu_surface_create(system, win, display);
+  }
+
+  getContext(context = "webgpu") {
```

Comment:
if we want to already be ready for OffscreenCanvas without having to do any unnecessary breaking changes, this should not provide a default value